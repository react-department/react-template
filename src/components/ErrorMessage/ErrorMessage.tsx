import { testIdErrorMessage } from '../../constants/TestId';

import type { ReactElement } from 'react';
import type IErrorMessage from './interfaces/IErrorMessage';

import styles from './sass/ErrorMessage.module.scss';

/**
 * Primary UI component to display error message for form fields.
 */
function ErrorMessage({ message }: IErrorMessage): ReactElement {
  return (
    <span className={styles.errorMessage} data-testid={testIdErrorMessage}>
      {message}
    </span>
  );
}

export default ErrorMessage;
