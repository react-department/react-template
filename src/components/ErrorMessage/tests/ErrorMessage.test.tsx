import { screen } from '@testing-library/react';

import { render } from '../../../../__tests__/test-utils';
import ErrorMessage from '../ErrorMessage';

describe('<ErrorMessage />', () => {
  test('should render component', () => {
    const { asFragment } = render(<ErrorMessage message="message" />);
    expect(asFragment()).toMatchSnapshot();
  });
  test('should render component with message', () => {
    render(<ErrorMessage message="message" />);
    expect(screen.getByText('message')).toBeInTheDocument();
  });
});
