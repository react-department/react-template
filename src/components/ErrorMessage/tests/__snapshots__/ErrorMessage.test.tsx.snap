// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage /> should render component 1`] = `
<DocumentFragment>
  <span
    class="errorMessage"
    data-testid="errorMessage"
  >
    message
  </span>
</DocumentFragment>
`;
